infrastructure/templates/public-api/types.bicep

```diff
@@ -64,14 +64,136 @@ type entraIdAuthenticationType = {
 }
 
 @export()
-type appGatewaySiteConfigType = {
-  resourceName: string
-  backendFqdn: string
-  backendDomainName: string
-  backendIpAddress: string
-  publicFqdn: string
+type AppGatewaySite = {
+  @description('Name of the site')
+  name: string
+
+  @description('Name of the certificate that applies for this site')
   certificateName: string
-  healthProbeRelativeUrl: string
+
+  @description('FQDN of the site')
+  fqdn: string
+}
+
+@export()
+type AppGatewayBackend = {
+  @description('Name of the backend')
+  name: string
+
+  @description('FQDN of the backend')
+  fqdn: string
+
+  @description('Path that the health probe should periodically ping e.g. /')
+  healthProbePath: string
+}
+
+@export()
+type AppGatewayRoute = {
+  name: string
+
+  @description('Name of the site')
+  siteName: string
```

Comment:
Is this a generic name, or is it the name of the target Azure resource itself?  Might be worth adding the distinction into the param descriptions now that variables have been renamed in some places that previously were more explicit (e.g. "resourceName" changing to "name" in some places now makes it a bit less obvious if something is just a descriptive name or if it's more defined than that).

Reply:
Nice one!